src/Build/Evaluation/Expander.cs

```diff
@@ -3316,10 +3317,77 @@ internal object Execute(object objectInstance, IPropertyProvider<T> properties,
             /// <returns>True if the well known function call binding was successful</returns>
             private bool TryExecuteWellKnownFunction(out object returnVal, object objectInstance, object[] args)
             {
+                returnVal = null;
+
                 if (objectInstance is string)
                 {
                     string text = (string)objectInstance;
```

Comment:
opportunity for 

```c#
if (objectInstance is string text)
```

Reply:
My plan is to take this PR first, and rebase and re-adjust #3502, which does exactly this.